### DISCUSSION TITLE
`Telescope repo cached_list` unusable on Mac OS

### DISCUSSION THREAD

**Discussion by cljoly** [cljoly/telescope-git-repo.nvim#13] [General]
When the use of the locate command was [initially introduced](https://github.com/cljoly/telescope-repo.nvim/pull/9), we have not been able to confirm that the locate command worked. @molleweide kindly volunteered to check [this particular message](https://github.com/cljoly/telescope-repo.nvim/pull/9#issuecomment-984144394). The purpose of this thread is to continue the discussion started in the [PR](https://github.com/cljoly/telescope-repo.nvim/pull/9) to get `locate` to work on Mac OS.

As I don’t have a Mac to test on, I used this [GitHub action workflow](https://github.com/cljoly/telescope-repo.nvim-macos/blob/1a186a9a378bd48e92fc5326c5ad6889d61b574f/.github/workflows/blank.yml) to get the extension to work on a bare bone system. So the following command executed on a setup similar to the workflow should yield a list of repositories on the system:
```
nvim +'lua require"telescope".extensions.repo.cached_list{bin="glocate", locate_opts={"-d", "/Users/runner/locatedb"}}'
```
![image](https://user-images.githubusercontent.com/7347374/144329067-46942bfd-736b-43bd-a4f9-928f8f3d878f.png)

Anyone feel free to report problems using the `cached_list` command here. Thanks!

**EDIT**: I’ve added support for `:checkhealth telescope`, the output might be further indication of anything going wrong.

**Comment by molleweide** [comment]
Yo!!!

Now I believe that I am getting some more time to work on repo and test.

I have to get `glocate` to work on my m1 macbook air.
The regular list command is really fast on m1.

**Comment by molleweide** [comment]
This is what checkhealth shows.
``` vim

telescope._extensions.repo: require("telescope._extensions.repo.health").check()
========================================================================
- WARNING: Install `glow` for a better preview of markdown files
- OK: Will use `bat` to preview markdown READMEs
- OK: Will use `bat` to preview non-markdown READMEs
- OK: locate: found `glocate`
locate (GNU findutils) 4.8.0
Packaged by Homebrew
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Eric B. Decker, James Youngman, and Kevin Dalley.
- ERROR: `:Telescope repo cached_list` was unsucessful. Exit code: 1
glocate -r /\.git$ -l 2

- OK: fd: found `fd`
fd 8.3.0
- INFO: Repos found for `:Telescope repo list`:
/Users/hjalmarjakobsson/code/plugins/nvim/telescope-repo.nvim...
```

**Comment by molleweide** [comment]
This is the command I had stored for loading the database to ~/locatedb.
Is there anything that stands out to you in it? And am I storing the DB in
the right location for it to work with `repo`?

``` bash
# https://egeek.me/2020/04/18/enabling-locate-on-osx/
if which glocate > /dev/null; then
  alias locate="glocate -d $HOME/locatedb"
  [[ -f "$HOME/locatedb" ]] && export LOCATE_PATH="$HOME/locatedb"
fi
alias loaddb="gupdatedb --localpaths=$HOME --prunepaths=/Volumes --output=$HOME/locatedb"
```

**Comment by molleweide** [comment]
IT WORKS!!!

The problem was that I forgot to reload shell after loading the database. I have to figure out a better way of doing this.
The `$LOCATE_PATH` variable needs to exist.

**Reply by cljoly** [reply]
Cool, thanks for figuring it out!

**Reply by molleweide** [reply]
Yeah now I'll be able to focus more on what you have done and help with new functionality.
